ql/src/java/org/apache/hadoop/hive/ql/optimizer/StatsOptimizer.java

```diff
@@ -299,7 +299,8 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx,
           Logger.info("Table " + tbl.getTableName() + " is external. Skip StatsOptimizer.");
           return null;
         }
-        if (MetaStoreUtils.isNonNativeTable(tbl.getTTable())) {
+        if (MetaStoreUtils.isNonNativeTable(tbl.getTTable())
```

Comment:
Please forgive the dumb question but in this case the isNonNativatTable could be anything else then Iceberg table?  Also  it would means the nonNative is those who has basicStats. Would not be better to have an util method IsIcebergTable(Table tbl) and inside the method check if it has basic stats. Outside of method to check is bit misleading  , but of course I do not know the full context.

Reply:
This is a more generic check. If non-native table can provide stats - we should use it (tbl.getStorageHandler().canProvideBasicStatistics()). 
At this moment only 'iceberg' has support for that, default is false.